GitUI/UserControls/RevisionGrid.cs

```diff
@@ -2383,20 +2383,24 @@ private void CherryPickCommitToolStripMenuItemClick(object sender, EventArgs e)
 
         private void FixupCommitToolStripMenuItemClick(object sender, EventArgs e)
         {
-            if (Revisions.RowCount <= LastRowIndex || LastRowIndex < 0)
+            if (ShouldStartDialog())
                 return;
 
             UICommands.StartFixupCommitDialog(this, GetRevision(LastRowIndex));
         }
 
         private void SquashCommitToolStripMenuItemClick(object sender, EventArgs e)
         {
-            if (Revisions.RowCount <= LastRowIndex || LastRowIndex < 0)
+            if (ShouldStartDialog())
                 return;
 
             UICommands.StartSquashCommitDialog(this, GetRevision(LastRowIndex));
         }
 
+        private bool ShouldStartDialog()
+        {
+            return Revisions.RowCount <= LastRowIndex || LastRowIndex < 0;
+        }
```

Comment:
I don't think it is needed.  Fixup already does the process of editing a
commit.  You just have to rebase afterwords using the interactive and
autosquash checkboxes.  Renaming is just a reword step in the rebase.

On Sun, Apr 10, 2016 at 7:13 AM lowdev <notifications@github.com> wrote:

> In GitUI/UserControls/RevisionGrid.cs
> <https://github.com/gitextensions/gitextensions/pull/3167#discussion_r59131866>
> :
>
> >                  return;
> >
> >              UICommands.StartSquashCommitDialog(this, GetRevision(LastRowIndex));
> >          }
> >
> > +        private bool ShouldStartDialog()
> > +        {
> > +            return Revisions.RowCount <= LastRowIndex || LastRowIndex < 0;
> > +        }
>
> missed new line
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/gitextensions/gitextensions/pull/3167/files/c5afb9f869b8792b8a3a25c3a927cc9a84287299#r59131866>
>